packs.go

```diff
@@ -37,44 +36,138 @@ func fileExists(filename string) bool {
 	return !info.IsDir()
 }
 
+// After a new pack download from upstream, update packs.json with installed packs
+func updatePacksInfo(langCode string, pack *Pack, packVersion *PackVersion) error {
+	packs, err := getPacksInfo()
+	if err != nil {
+		return err
+	}
+
+	var (
+		existingPackIndex int
+		existingPack      *Pack = nil
+	)
+
+	for index, packR := range packs {
+		if packR.Identifier == pack.Identifier {
+			existingPackIndex = index
+			existingPack = &packR
+			break
+		}
+	}
+
+	if existingPack == nil {
+		// will have one element
+		pack.Versions = []PackVersion{*packVersion}
+
+		// Append new pack
+		packs = append(packs, *pack)
+	} else {
+		// Append new pack version
+		existingPack.Versions = append(existingPack.Versions, *packVersion)
+
+		packs[existingPackIndex] = *existingPack
+	}
+
+	file, err := json.MarshalIndent(packs, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(getPacksInfoPath(), file, 0644)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
 // Download pack from upstream
-func downloadPackFile(langCode string, packVersionIdentifier string) (string, error) {
-	fileURL := fmt.Sprintf("%s/packs/%s/%s", varnamdConfig.upstream, langCode, packVersionIdentifier)
-	filePath := path.Join(getPacksDir(), langCode, "a"+packVersionIdentifier)
+func downloadPackFile(langCode string, packIdentifier string, packVersionIdentifier string) (*Pack, *PackVersion, string, error) {
```

Comment:
compact returns to two or three variables for better readability

Reply:
all of them are very necessary though, can't think of a way to reduce them to 3

Reply:
pack them to another struct and return